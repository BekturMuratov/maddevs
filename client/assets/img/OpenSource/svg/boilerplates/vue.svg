<svg xmlns="http://www.w3.org/2000/svg" width="336" height="125" fill="none" xmlns:v="https://vecta.io/nano"><g clip-path="url(#A)"><path d="M206.784.001l-38.797 67.203L129.191.001H-.008l167.996 290.997L335.984.001h-129.2z" fill="#41b883"/><path d="M206.785.001l-38.797 67.203L129.191.001h-62l100.798 174.595L268.786.001h-62.001z" fill="#34495e"/></g><defs><clipPath id="A"><path fill="#fff" d="M0 0h336v291H0z"/></clipPath></defs></svg>