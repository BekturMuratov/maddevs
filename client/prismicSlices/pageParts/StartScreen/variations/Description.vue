<template>
  <section
    data-testid="description-slice"
    class="description-slice"
  >
    <div class="container">
      <div class="description-slice__content">
        <h1
          data-testid="description-slice__title"
          class="description-slice__title"
          v-html="title"
        />
        <p
          v-if="subtitle"
          data-testid="description-slice__subtitle"
          class="description-slice__subtitle"
          v-html="subtitle"
        />
      </div>
    </div>
  </section>
</template>

<script>
export default {
  name: 'Description',

  props: {
    slice: {
      type: Object,
      required: true,
      default: () => ({}),
    },
  },

  data() {
    return {
      title: this.slice?.primary?.title,
      subtitle: this.slice?.primary?.subtitle,
    }
  },
}
</script>

<style lang="scss" scoped>
.description-slice {
  font-family: Inter;
  font-style: normal;

  &__content {
    padding: 80px;
    background: #1D1D1F;
    border-radius: 4px;
    margin-bottom: 50px;

    @media screen and (max-width: 480px) {
      padding: 32px;
    }
  }

  &__title {
    font-weight: 700;
    font-size: 56px;
    line-height: 65px;
    letter-spacing: -0.04em;
    color: #FFFFFF;
    margin-bottom: 40px;

    @media screen and (max-width: 700px) {
      font-size: 30px;
      line-height: 37px;
    }
  }

  &__subtitle {
    font-weight: 600;
    font-size: 24px;
    line-height: 38px;
    letter-spacing: -0.013em;
    color: #FFFFFF;

    @media screen and (max-width: 700px) {
      font-size: 20px;
      line-height: 30px;
    }
  }
}
</style>
