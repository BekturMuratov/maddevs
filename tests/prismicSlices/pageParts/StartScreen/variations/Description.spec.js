import { render, screen } from '@testing-library/vue'
import Description from '@/prismicSlices/pageParts/StartScreen/variations/Description'
import { getFakePropsByParams } from '../testsUtils'

const apiData = {
  title: 'Description title',
  subtitle: 'Description subtitle',
}

describe('Description slice', () => {
  it('should render correctly with data', () => {
    const { container } = render(Description, {
      props: getFakePropsByParams(apiData),
    })

    expect(screen.getByText(apiData.title)).not.toBeNull()
    expect(screen.getByText(apiData.subtitle)).not.toBeNull()
    expect(container).toMatchSnapshot()
  })

  it('should render correctly without data', () => {
    const { container } = render(Description, {
      props: getFakePropsByParams({}),
    })

    expect(screen.queryByTestId('description-slice__title')).not.toBeNull()
    expect(screen.queryByTestId('description-slice__subtitle')).toBeNull()
    expect(container).toMatchSnapshot()
  })
})
