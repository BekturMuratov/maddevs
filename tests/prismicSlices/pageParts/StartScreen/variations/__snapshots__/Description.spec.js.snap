// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Description slice should render correctly with data 1`] = `
<div>
  <section
    class="description-slice"
    data-testid="description-slice"
  >
    <div
      class="container"
    >
      <div
        class="description-slice__content"
      >
        <h1
          class="description-slice__title"
          data-testid="description-slice__title"
        >
          Description title
        </h1>
         
        <p
          class="description-slice__subtitle"
          data-testid="description-slice__subtitle"
        >
          Description subtitle
        </p>
      </div>
    </div>
  </section>
</div>
`;

exports[`Description slice should render correctly without data 1`] = `
<div>
  <section
    class="description-slice"
    data-testid="description-slice"
  >
    <div
      class="container"
    >
      <div
        class="description-slice__content"
      >
        <h1
          class="description-slice__title"
          data-testid="description-slice__title"
        />
         
        <!---->
      </div>
    </div>
  </section>
</div>
`;
